picard/tagger.py

```diff
@@ -358,17 +359,45 @@ def add_files(self, filenames, target=None):
             if filename not in self.files:
                 file = open_file(filename)
                 if file:
-                    self.files[filename] = file
-                    new_files.append(file)
-        if new_files:
+                    tmp_files[filename] = file
+        if tmp_files and self.check_load(tmp_files):
+            new_files = []
+            for filename in sorted(tmp_files):
+                file = tmp_files[filename]
+                self.files[filename] = file
+                new_files.append(file)
             log.debug("Adding files %r", new_files)
-            new_files.sort(key=lambda x: x.filename)
             if target is None or target is self.unmatched_files:
                 self.unmatched_files.add_files(new_files)
                 target = None
             for file in new_files:
                 file.load(partial(self._file_loaded, target=target))
 
+    def check_load(self, new_files):
+        #Load large amounts of files
+        num_files = len(new_files)
+        if num_files > config.setting["file_import_threshold"]:
+            QMessageBox = QtGui.QMessageBox
+
+            msg = QMessageBox()
+            msg.setIcon(QMessageBox.Warning)
+            msg.setWindowModality(QtCore.Qt.WindowModal)
+            msg.setWindowTitle(_(u"Large File Load"))
```

Comment:
Title of the dialog should indicate it is a warning, "Large File Load" looks unrelated.
What about "Warning: too many files at once" ? or something like that.